labs/b10.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Lab 10 - Version Control and Backups
+layout: lab
+---
+
+## Overview
+
+In this lab, we'll be talking about using version control, in particular Git, for keeping track of changes to your files and backing up your files to other machines. 
+We'll be setting up [Gitolite](http://gitolite.com/gitolite/index.html), an open-source Git server that allows you to host Git repositories on other machines,
+and set access to Git repositories hosted on the Gitolite server.
+
+## A Git Primer
+
+Version control tools are designed to keep track of changes to files, and allow you to rewind to some last saved state. 
+git, as one of the most popular version control tools in use today, supports these features, but also allows for more advanced workflows such as splitting a filebase into "branches" for experimentation.
+In order to complete this lab, you'll need to know a few basic git commands and some terminology.
+
+- A folder can be made into a git **repository** by running `git init`, telling git that files under the folder can start being tracked by git.
+- A git repository can also be **cloned** from online by running `git clone [repository url]`, which will create a copy of the repository
+on your machine.
+- However, files do not have their changes tracked by git by default. In order to tell git to track changes made to a file, run `git add [file]`.
+- git groups a set of changes into a **commit**. Tracked files in a git repository do not have their contents saved in git until `git commit` is run. You can think of `git commit` as creating a snapshot of all modifications made to all tracked files since the last commit.
+- In order to back up changes in a repository to another machine, or download changes from another machine to your local machine, git allows you to set **remotes**. Adding a remote is done by running `git remote add [remote url]`.
+- In order to back up changes on a local repository to a remote, run `git push [remote name]`. By convention, the primary remote set for a repository is called "origin". If you clone a repository, its origin is automatically set to the URL you cloned from.
+- In order to download changes from a remote to a local repository, run `git pull [remote name]`.
+
+## Gitolite
+
+### Generating an SSH key
+
+Gitolite identifies its users through their SSH public key. Since you'll need to interact with Gitolite from your OCF account on tsunami, in
+order to complete the lab, you'll first have to generate an SSH keypair on tsunami.
+
+On `tsunami.ocf.berkeley.edu`, create a directory called `.ssh` in your home directory. Within `.ssh`, run `ssh-keygen` and hit enter until
```

Comment:
maybe something here in case students already have `.ssh` for some reason so they don't get confused